adls-plugins/src/main/java/co/cask/hydrator/plugin/action/ADLSDeleteAction.java

```diff
@@ -0,0 +1,162 @@
+package co.cask.hydrator.plugin.action;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.action.Action;
+import co.cask.cdap.etl.api.action.ActionContext;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * ADLS delete action plugin to delete files on adls
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name("ADLSDelete")
+@Description("Action to delete files on ADLS")
+public class ADLSDeleteAction extends Action {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ADLSDeleteAction.class);
+
+  private ADLSDeleteActionConfig config;
+  private PathFilter filter;
+
+  public ADLSDeleteAction(ADLSDeleteActionConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    Path path = new Path(config.path);
+
+    Configuration conf = new Configuration();
+
+    Map<String, String> properties = getFileSystemProperties();
+    for (Map.Entry<String, String> entry : properties.entrySet()) {
+      conf.set(entry.getKey(), entry.getValue());
+    }
+
+    FileSystem fileSystem = FileSystem.get(conf);
+
+    FileStatus[] listFiles;
+    if (config.fileRegex != null) {
+      PathFilter filter = new PathFilter() {
+        private final Pattern pattern = Pattern.compile(config.fileRegex);
+
+        @Override
+        public boolean accept(Path path) {
+          return pattern.matcher(path.getName()).matches();
+        }
+      };
+      listFiles = fileSystem.listStatus(path, filter);
+    } else {
+      listFiles = fileSystem.listStatus(path);
+    }
+
+    for (FileStatus file: listFiles) {
+      Path currPath = file.getPath();
+      removePath(fileSystem, currPath);
+    }
+
+    if (fileSystem.isDirectory(path) && config.fileRegex == null) {
```

Comment:
seems like this should happen before listing getting listFiles, with file listing logic skipped if this is true

Reply:
But we will have to delete directory after deleting all the files in side it right?

Reply:
doesn't `fileSystem.delete(currPath, true)` delete the directory as well? Just seems like we should just remove the directory instead of listing everything in the directory, deleting those, then deleting the directory.